### ISSUE TITLE
[A11ySev2] [Screen reader – Dashboards with Grafana – Azure Insights Key Vaults-Queries]: Screen reader announces blank for options under the Data Source combo box in the Queries tab.

### ISSUE THREAD

**Issue by LinglingTong** [grafana/grafana#118769]
## Steps to reproduce

Repro Steps: 
  
1. Open 'Azure Monitor dashboards with Grafana' in Azure Portal
2. "Monitor | Dashboards with Grafana | Gallery" screen appears with already Dashboards with Grafana tab selected.  
3. Select already existing data as 'Azure Insights Key Vaults'  under All Dashboards tab.  
4. Navigate to toolbar and select edit button. Now select Add Visualization from Add menu. 
5. Navigate all the controls under Queries tab. 
6. Observe that the screen reader does not announce a label for the combo box, making it unclear what the combo box is for. 

**Actual Result:**
The screen reader does not announce any label for the service combo box under the Queries tab, leaving users unaware of its purpose.  
**Observation**:  
The service combo box does not have an accessible name or label. This results in the screen reader failing to announce its purpose, which can confuse users relying on assistive technologies.  **Expected Result: ** 
The service combo box should have a programmatically associated label that clearly describes its purpose. Screen reader should announce the 'service' label for the combo box.

**Relevant WCAG Criteria:** [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) (Level A)

## Environment

- **Grafana version:** 12.2.0
- **Data source type & version:**
- **User OS & Browser:** Version 25H2 (OS Build 26200.7093) / Chrome Version 142.0.7444.177 (Official Build) (64-bit)
- **Others:**

**Comment by LinglingTong** [comment]
<img width="1919" height="1041" alt="Image" src="https://github.com/user-attachments/assets/35a1baf7-290c-460d-bca0-e3e220262e07" />

**Comment by v-kavyagupta** [comment]
Hi @LinglingTong 
We verified this issue on the Grafana nighty versionv 13.2.0-28412642318 (6b36314bd9)  http://localhost:3000/d/adq6vrs/new-dashboard?orgId=1&from=now-1h&to=now&timezone=browser&editPanel=3
Issue is still repro, Screen reader announces blank for options under the Data Source combo box in the Queries tab.
Attaching the screenshot for the reference.
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/b92a35d7-97db-4396-bf3b-7bd0d71f0a79" />

**Comment by v-kavyagupta** [comment]
We verified this issue on the URL:http://localhost:3000/d/adq6vrs/new-dashboard?orgId=1&from=now-1h&to=now&timezone=browser&editPanel=5
Issue is still repro, Screen reader announces blank for options under the Data Source combo box in the Queries tab.
 
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/514699c5-5865-4784-b568-3c62fa669323" />

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/3f3cf56b-10f3-4a26-a707-c46c4d093142" />

**Comment by ashharrison90** [comment]
https://github.com/user-attachments/assets/e4187f8e-5b87-46fe-b812-c4420b0f85c0

hmmmmm, i wonder if this is a problem with NVDA? here's voiceover showing the correct behaviour 🤔
